### ISSUE TITLE
Use PROJECT_SOURCE_DIR instead of CMAKE_SOURCE_DIR

### ISSUE THREAD

**Issue by cwansart** [ogdf/ogdf#22]
Hello,

I am trying to use ogdf as a subproject of mine. So I added it via `add_subdirectory` to my project. It had some cmake issues because you use CMAKE_SOURCE_DIR. It tried to find files in my root directory instead of its `lib/ogdf` subdirectory.

Some googling got me to this: https://stackoverflow.com/questions/22214349/modify-cmake-source-dir
They recommend to use PROJECT_SOURCE_DIR instead.

The files that are affected are:
 * ogdf/CMakeLists.txt line 4
 * ogdf/cmake/coin.cmake lines 17 and 20.

I have not checked if CMAKE_SOURCE_DIR is used somewhere else since ogdf compiled without any issues.

**Comment by TiloW** [comment]
Thanks for reporting!

We will switch to `PROJECT_SOURCE_DIR`.

**Comment by cwansart** [comment]
Thank you.

**Comment by schulzch** [comment]
Switching to PROJECT_SOURCE_DIR is a good idea, but to add OGDF (and other externals) better use `ExternalProject_Add`, e.g.:
```CMake
# Configure
include(ExternalProject)
ExternalProject_Add(OGDF_PROJECT
    DOWNLOAD_COMMAND ""
    UPDATE_COMMAND ""
    INSTALL_COMMAND ""
    SOURCE_DIR "${CMAKE_SOURCE_DIR}/lib/ogdf"
    CMAKE_ARGS "-DOGDF_MEMORY_MANAGER=MALLOC_TS -DOGDF_DEBUG=OFF")
ExternalProject_Get_Property(OGDF_PROJECT install_dir)
ExternalProject_Get_Property(OGDF_PROJECT source_dir)
add_library(OGDF STATIC IMPORTED)
set_property(TARGET OGDF
    PROPERTY IMPORTED_LOCATION "${install_dir}/src/OGDF_PROJECT-build/${CMAKE_CFG_INTDIR}/OGDF.lib")
add_library(COIN STATIC IMPORTED)
set_property(TARGET COIN
    PROPERTY IMPORTED_LOCATION "${install_dir}/src/OGDF_PROJECT-build/${CMAKE_CFG_INTDIR}/COIN.lib")
include_directories("${source_dir}/include" "${install_dir}/src/OGDF_PROJECT-build/include")

# Target
target_link_libraries(your_target OGDF COIN)
```

**Comment by sbeyer** [comment]
Fixed by the latest snapshot commit acd58c7
